web/source/kmwosk.ts

```diff
@@ -213,7 +214,7 @@ namespace com.keyman {
 
     getId(): string {
       // Define each key element id by layer id and key id (duplicate possible for SHIFT - does it matter?)
-      return this.spec.layer+'-'+this.spec.id;
+      return this.layer+'-'+this.spec.id;
```

Comment:
Is it possible for `this.layer` to be undefined?

Can the constructor in l. 31 initialize?
```
this.layer = 'default';
```